cli/compilers/ts.rs

```diff
@@ -468,15 +511,23 @@ impl TsCompiler {
     module_specifier: &ModuleSpecifier,
     contents: &str,
   ) -> std::io::Result<()> {
+    let source_file = self
+      .file_fetcher
+      .fetch_cached_source_file(&module_specifier)
+      .expect("Source file not found");
+
+    // NOTE: JavaScript files are only cached to disk if `checkJs`
+    // option in on
+    if source_file.media_type == msg::MediaType::JavaScript && !self.compile_js
```

Comment:
What would happen if you removed the `!self.compile_js` condition? 

It doesn't seem helpful to place a copy of JS in the DENO_DIR...

Reply:
I'm not sure I understand the question; I think when you type check JS then TS compiler can emit different output that provided source code.

Reply:
Probably it can change depending on the target type (e.g. ESM to AMD) but I'm not sure that's relevant for us. There should be no extra information in the outputted JS.

Reply:
I'm happy to remove that, but only after problematic parts are rewritten. I don't want to introduce logic changes in these refactors.

Reply:
SGTM